include/model/CAnomalyScore.h

```diff
@@ -175,40 +173,72 @@ class MODEL_EXPORT CAnomalyScore {
 
         //! \name Serialization
         //@{
-        //! Persist state by passing information to the supplied inserter
+        //! Persist state by passing information to \p inserter.
         void acceptPersistInserter(core::CStatePersistInserter& inserter) const;
 
-        //! Create from a state document.
+        //! Initialize reading state from \p traverser.
         bool acceptRestoreTraverser(core::CStateRestoreTraverser& traverser);
         //@}
 
-    public:
         //! Get a checksum of the object.
         uint64_t checksum() const;
 
     private:
         using TDoubleDoublePr = std::pair<double, double>;
         using TDoubleDoublePrVec = std::vector<TDoubleDoublePr>;
+        //! \brief Wraps a maximum score.
+        class CMaxScore {
+        public:
+            //! Update with \p score.
+            void add(double score);
+
+            //! Get the score.
+            double score() const;
+
+            //! Age the score.
+            void age(double alpha);
+
+            //! Check if we should delete this score given elapsed \p time.
+            bool forget(double time);
+
+            //! Persist state by passing information to \p inserter.
+            void acceptPersistInserter(core::CStatePersistInserter& inserter) const;
+
+            //! Initialize reading state from \p traverser.
+            bool acceptRestoreTraverser(core::CStateRestoreTraverser& traverser);
+
+            //! Get a checksum of the object.
+            uint64_t checksum() const;
+
+        private:
+            TMaxValueAccumulator m_Score;
+            double m_TimeSinceLastScore;
+        };
+        using TWordMaxScoreUMap = TDictionary::CWordUMap<CMaxScore>::Type;
 
     private:
         //! Used to convert raw scores in to integers so that we
         //! can use the q-digest.
-        static const double DISCRETIZATION_FACTOR;
+        constexpr static const double DISCRETIZATION_FACTOR = 1000.0;
```

Comment:
Do you need both `const` and `constexpr`?  I would have thought `constexpr` implies `const`.

Also, since we used to write `const` after `static`, should we also write `constexpr` after `static` since it's like a stronger form of `const`?